src/parser/core/modules/BuffWindow.tsx

```diff
@@ -219,7 +219,13 @@ export abstract class BuffWindowModule extends Module {
 
 			if (windowDurationMillis >= fightTimeRemaining) {
 				const gcdEstimate = this.globalCooldown.getEstimate()
-				return Math.ceil((windowDurationMillis - fightTimeRemaining) / gcdEstimate)
+				const newExpectedGCDs = Math.ceil((windowDurationMillis - fightTimeRemaining) / gcdEstimate)
+
+				if ( this.expectedGCDs ) {
+					if ( newExpectedGCDs <= this.expectedGCDs.expectedPerWindow ) {
+						return newExpectedGCDs
```

Comment:
This is going to result in not allowing any reduction in expected GCDs for a situation like "used Meikyo with 1.5 seconds left in the fight", since the duration vs gcdEstimate is coming up with 4 expected GCDs to forgive.

Also, naming this variable `newExpectedGCDs` is wrong - what's being calculated is the number of GCDs to remove, not what the expected amount per the window is (getBuffWindowExpectedGCDs subtracts the result of this function from the baseline expectation per window, in addition to adding the values the class overrides provide, if any).

I think capping the overall result of getBuffWindowExpectedGCDs to a minimum of 0 or 1 may make some sense, but it's probably better for classes with "non-standard" buffs like Acceleration or Meikyo (whose duration is much longer than the number of casts required to consume all the stacks and early-end the buff) to override reduceExpectedGCDsEndOfFight directly with logic that properly accounts for their implicitly shorter "actual" duration.

Reply:
Point Taken, I'll close this PR and just redo the logic on a override using just the fight duration left / estimatedGCD